main.lua

```diff
@@ -651,56 +686,72 @@ local function isCtrlOrGuiDown()
-			gif_recording, err=gif.new(cartname..'-'..os.time()..'.gif')
-			if not gif_recording then
-				log('failed to start recording: '..err)
-			else
-				gif_canvas=love.graphics.newCanvas(pico8.resolution[1]*2, pico8.resolution[2]*2)
-				log('starting record ...')
-			end
-		else
-			log('recording already in progress')
-		end
-
-	elseif key=='f4' or key=='f9' then
-		-- stop recording and save
-		if gif_recording~=nil then
-			gif_recording:close()
-			log('saved recording to '..gif_recording.filename)
-			gif_recording=nil
-			gif_canvas=nil
-		else
-			log('no active recording')
-		end
-
-  elseif cart and pico8.cart._keydown then
-    return pico8.cart._keydown(key)
+  log(key)
+  if paused then
+    if key=='z' or key=='x' or key=='c' or key=='v' or key=='return' then
+      local handler = paused_menu[paused_selected+1][2]
+      if handler then handler() end
+    elseif key=='up' then
+      paused_selected=(paused_selected-1)%4
+    elseif key=='down' then
+      paused_selected=(paused_selected+1)%4
+    elseif key=='pause' or key=='p' then
+      paused=false
+    end
+  else
+    if key=='r' and isCtrlOrGuiDown() then
+      api.music()
+      _load()
+    elseif key=='q' and isCtrlOrGuiDown() then
+      love.event.quit()
+    elseif key=='v' and isCtrlOrGuiDown() then
+      pico8.clipboard=love.system.getClipboardText()
+    elseif key=='pause' or key=='p' or key=='return' then
+      local function p_label()
+        if muted then return "off" else return "on" end
+    end
+    paused=true
+    paused_selected = 0
+    paused_menu={
+      {"continue", function() paused=false end},
+      {"----", nil},
+      {"sound "..p_label(), function() muted=not muted paused_menu[3][1]="sound "..p_label() end},
+      {"reset cart", function() api.music() _load() end}
+    }
+  	elseif key=='m' and isCtrlOrGuiDown() then
+		  muted=not muted
+    elseif key=='f1' or key=='f6' then
+      -- screenshot
+      local filename=cartname..'-'..os.time()..'.png'
+      love.graphics.captureScreenshot(filename)
+      log('saved screenshot to', filename)
+    elseif key=='f3' or key=='f8' then
+      -- start recording
+      if gif_recording==nil then
+        local err
+        gif_recording, err=gif.new(cartname..'-'..os.time()..'.gif')
+        if not gif_recording then
+          log('failed to start recording: '..err)
+        else
+          gif_canvas=love.graphics.newCanvas(pico8.resolution[1]*2, pico8.resolution[2]*2)
+          log('starting record ...')
+        end
+      else
+        log('recording already in progress')
+      end
+    elseif key=='f4' or key=='f9' then
+      -- stop recording and save
+      if gif_recording~=nil then
+        gif_recording:close()
+        log('saved recording to '..gif_recording.filename)
+        gif_recording=nil
+        gif_canvas=nil
+      else
+        log('no active recording')
+      end
+    elseif cart and pico8.cart._keydown then
+      return pico8.cart._keydown(key)
+    end
```

Comment:
Major restructuring of this to support interactive pause menu and prepare to support user defined custom menus